### PR TITLE
[bitnami/rabbitmq] Fix volume-permissions initContainer if runAsUser is auto

### LINKED ISSUE #8932: [bitnami/rabbitmq] OpenShift: volume-permissions ignore .Values.volumePermissions.securityContext.runAsUser="auto"
### Name and Version

bitnami/rabbitmq 8.28.1

### What steps will reproduce the bug?

OpenShift which has scc with dynamic userids (and 0 is not allowed). We should be able to use another User for volume-permissions container.
We are not abot today to tif those rules because of the following line:

https://github.com/bitnami/charts/blob/6fd94c4b52c442a1e792a5d11a753135c5199d4e/bitnami/rabbitmq/templates/statefulset.yaml#L88

Here my values, but I'm unable to run inside OpenShift Container

### Are you using any custom parameters or values?

```yaml
podSecurityContext:
  enabled: false
containerSecurityContext: null
volumePermissions:
  enabled: true
  securityContext:
    runAsUser: "auto"
```

### What is the expected behavior?

Working on OpenShift

### What do you see instead?

The container are not starting:
```
create Pod fleet-executer-dev-rabbitmq-0 in StatefulSet fleet-executer-dev-rabbitmq failed error: pods "fleet-executer-dev-rabbitmq-0" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, spec.initContainers[0].securityContext.runAsUser: Invalid value: 0: must be in the ranges: [1000740000, 1000749999], provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]
```
### Additional information

I'm able to run MongoDB from bitnami in OpenShift, thanks to the following code:

https://github.com/bitnami/charts/blob/6fd94c4b52c442a1e792a5d11a753135c5199d4e/bitnami/mongodb/templates/standalone/dep-sts.yaml#L113

Is something that ca be fixed please ?
Thanks a lot

### LINKED ISSUE #9984: RabbitMQ cannot start on Openshift without manipulating volumePermission initContainer runAsUser
### Name and Version

bitnami/rabbitmq

### What steps will reproduce the bug?

Helm install current RabbitMQ Chart below OCP at least 4.7+
with
 
```yaml
volumePermissions.enabled=true
podSecurityContext.enabled=true 
  runAsUser: 1000800000
  fsGroup: 1000800000
```

### What is the expected behavior?

The pod should be scheduled

### What do you see instead?

Pod cannot be scheduled even with `securityContext.enabled=true` as `fsGroup` is not being picked by the `initContainer` for `volumePermissions`

### Additional information

please refer https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/templates/statefulset.yaml#L98 

why do we not use `.Values.podSecurityContext.runAsUser` in that case aswell?

### PR DESCRIPTION
**Pull request by link-webcreations**
### Description of the change

This make **volume-permissions** initContainer able to run without error if value `volumePermissions.containerSecurityContext.runAsUser=auto`

### Benefits

OpenShift users will be able to run this initContainer with an arbitrary user id.

### Possible drawbacks

Not known. the default behaviour remain the same. It just fix the use case of the "auto" user id.

### Applicable issues

  - fixes #9984
  - fixes #8932

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/rabbitmq/templates/statefulset.yaml [hunk 1]
```diff
@@ -92,8 +92,17 @@ spec:
             - -ec
             - |
               mkdir -p "{{ .Values.persistence.mountPath }}"
+              {{- if eq ( toString ( .Values.volumePermissions.containerSecurityContext.runAsUser )) "auto" }}
+              chown -R `id -u`:`id -G | cut -d " " -f2` {{ .Values.persistence.mountPath }}
```
Comment (joancafom) [inline bitnami/rabbitmq/templates/statefulset.yaml:96]: This first command should not be recursive, and only change the ownership of the folder. Equivalent to line 95:

https://github.com/bitnami/charts/blob/74e05c5c3e31b53508cba7365388b6710ba9c854/bitnami/rabbitmq/templates/statefulset.yaml#L95

**Review by joancafom — CHANGES_REQUESTED** [review]
Thanks for your PR @link-webcreations! I left a comment, could you please review it?

File: bitnami/rabbitmq/templates/statefulset.yaml (hunk 1 above)
Comment (link-webcreations) [reply bitnami/rabbitmq/templates/statefulset.yaml]: Thanks! updated.

**Comment by link-webcreations** [comment]
> Thanks for your PR @link-webcreations! I left a comment, could you please review it?

Done :)

**Review by joancafom — APPROVED** [review]
LGTM! Thanks for your changes 🚀

**Pull request merged**